twine/auth.py

```diff
@@ -57,9 +70,65 @@ def password(self) -> Optional[str]:
             self.input.password,
             self.config,
             key="password",
-            prompt_strategy=self.password_from_keyring_or_prompt,
+            prompt_strategy=self.password_from_keyring_or_trusted_publishing_or_prompt,
         )
 
+    def make_trusted_publishing_token(self) -> Optional[str]:
+        # Trusted publishing (OpenID Connect): get one token from the CI
+        # system, and exchange that for a PyPI token.
+        repository_domain = cast(str, urlparse(self.system).netloc)
+        session = requests.Session()  # TODO: user agent & retries
```

Comment:
Did you forget this TODO or are you leaving it out of this PR? Would be nice if this had the same user-agent that is used here: https://github.com/pypa/twine/blob/66b25ae804b7b5f7cde4ef3fab0c262bcc1a84f0/twine/repository.py#L79-L85

Reply:
Good catch, I had forgotten that. I've moved the common session setup to a function in utils now, so it can be used both here and in the repository module.